### DISCUSSION TITLE
PostgREST cannot find any function, although it's listed in paths object

### DISCUSSION THREAD

**Discussion by ohrie** [PostgREST/postgrest#2495] [Q&A] [answered]
Hey there!

I've setup PostgREST 10.0.0 with swagger and it works very well for tables and views. But I don't get the functions working. I have PostGIS enabled, which provides a bunch of functions. Running for example `localhost/rpc/postgis_version` returns the following:

```json
{
    "code": "PGRST202",
    "details": null,
    "hint": "If a new function was created in the database with this name and parameters, try reloading the schema cache.",
    "message": "Could not find the public.postgis_version function with a single json or jsonb parameter in the schema cache"
}
```

As told in the response, I've restarted the PostgREST docker compose service to recreate the schema cache. This doesn't change anything.
Although it's in the pahts object listed like this:

```json
"/rpc/postgis_version": {
      "post": {
        "tags": ["(rpc) postgis_version"],
        "summary": "Returns PostGIS version number and compile-time options.",
        "produces": ["application/json", "application/vnd.pgrst.object+json"],
        "parameters": [
          {
            "in": "body",
            "name": "args",
            "required": true,
            "schema": {
              "description": "Returns PostGIS version number and compile-time options.",
              "type": "object"
            }
          },
          { "$ref": "#/parameters/preferParams" }
        ],
        "responses": { "200": { "description": "OK" } }
      }
    },
```

It tells me, that I have to add a body, but the SQL function does not require such.
Same happens with own function wrote and saved. The role has `EXECUTE` permissions.
Does somebody has an idea why this happens? As I told earlier: Views and Tables work as expected.

Thank you!

**Comment by steve-chavez** [comment, marked as answer]
Hello,

> Running for example localhost/rpc/postgis_version returns the following

How are you making the request? (curl snippet)
